MiniApp/Classes/MiniAppLister.swift

```diff
@@ -1,2 +1,54 @@
 internal class MiniAppLister {
+    var environment: Environment
+    var miniAppClient: MiniAppClient
+
+    init(environment: Environment) {
+        self.environment = environment
+        self.miniAppClient = MiniAppClient()
+    }
+
+    func fetchList(completionHandler: @escaping (Result<[MiniAppInfo], Error>) -> Void) {
+        guard let url = environment.listingUrl else {
+            return completionHandler(.failure(self.invalidURLError()))
+        }
+
+        self.miniAppClient.requestFromServer(request: listingURLRequest(url: url)) { (result) in
+            switch result {
+            case .success(let responseData):
+                guard let decodeResponse = self.decodeListingResponse(with: responseData.data) else {
+                    return completionHandler(.failure(self.invalidResponseData()))
+                }
+                return completionHandler(.success(decodeResponse))
+            case .failure(let error):
+                return completionHandler(.failure(error))
+            }
+        }
+    }
+
+    func listingURLRequest(url: URL) -> URLRequest {
+        var urlRequest = URLRequest(url: url)
+        urlRequest.setAuthorizationHeader(environment: environment)
+        return urlRequest
+    }
+
+    func decodeListingResponse(with dataResponse: Data?) -> [MiniAppInfo]? {
+        do {
+            return try JSONDecoder().decode(Array<MiniAppInfo>.self, from: dataResponse!) as [MiniAppInfo]
```

Comment:
Don't force unwrap. Use a guard at the top of the function.

Reply:
Done